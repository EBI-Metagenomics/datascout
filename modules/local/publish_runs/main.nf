#!/usr/bin/env nextflow
process PUBLISH_RUNS {

    debug true
    publishDir "${params.output}", mode: "copy"
    label 'process_low'

    input:
      tuple val(meta), path(fastq_files)

    output:
      tuple val(meta), path("${meta.genome_id}_${meta.ena_tax}_rna_fastq_dir/*"), emit: fastq_files

    script:
    """
     echo "publishing fastq files for ${meta.genome_id}_${meta.ena_tax}"
    """
}

// Publish filtered fastq files only 
