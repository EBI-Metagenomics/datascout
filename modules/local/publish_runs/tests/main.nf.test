nextflow_process {

    name "Test Process PUBLISH_RUNS"
    script "modules/local/publish_runs/main.nf"
    process "PUBLISH_RUNS"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = tuple([id:"GCA_003719455.1", ena_tax:"default"], [ "$projectDir/tests/reference_databases/SRRtest_1.fastq", "$projectDir/tests/reference_databases/SRRtest_2.fastq"] )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.fastq_files).match()
        }

    }

}
