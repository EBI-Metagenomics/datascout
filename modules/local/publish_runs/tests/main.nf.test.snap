{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    [
                        "/Users/vkale/Downloads/pipelines/datascout/.nf-test/tests/2e25632c12fcffa933fe289bb77958c3/work/b6/1eb6804248a35e85b0d1c5d231e86c/GCA_003719455.1_default_rna_fastq_dir/SRRtest_1.fastq",
                        "/Users/vkale/Downloads/pipelines/datascout/.nf-test/tests/2e25632c12fcffa933fe289bb77958c3/work/b6/1eb6804248a35e85b0d1c5d231e86c/GCA_003719455.1_default_rna_fastq_dir/SRRtest_2.fastq"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-09T13:40:35.558922"
    }
}